### PR TITLE
Bug/repair game draw step and upload docs warnings

### PR DESCRIPTION
**Pull request by OniWarrior**
In the gameDrawStep.js component there were two warnings that needed to be addressed.
 One was a useEffect dependancy warning and the other was a alt text warning on the image element. The useEffect warning was the most important because there was a missing argument in the dependency array on line 100 that was causing performance issues for the application. I added the missing argument of the prop for the file submission drawing and performance has increased significantly. The alt text warning on line 106 was minor in that the text inside the string was redundant in its wording . I made a slight change to the text to repair the warning. 

In the UploadDocs component, there was a missing dependency list for the dependency array on line 121. I added the 3 missing arguments to repair the warning.  The next warning involved improper syntax for a condition on line 117. This was minor; I corrected the syntax and the warning disappeared.

### CODE REVIEW

**Comment by OniWarrior** [comment]
Note: While my changes repaired the warnings and increased performance. There was an unintended side effect. This side effect is minor. The modal window that should appear  when GameReadStep component initially mounts does not appear. I suspect that when the other warnings are repaired that this won't be the case anymore.

**Review by ashtilawat — CHANGES_REQUESTED** [review]
Looks like the package-lock was committed here. We can't have the package.json or package-lock being committed to the repo. I'm guessing we did a "git add .". Let's only add the necessary files by doing a git add fileName.

**Comment by OniWarrior** [comment]
Apologies on the package-lock.json file. I deleted the commited file.

File: src/components/pages/Gameification/GameDrawStep.js [hunk 1]
```diff
@@ -97,13 +97,13 @@ export default function GameDrawStep(props) {
...
     setFileList(props.fileSubmissionData.drawings);
-  }, []);
+  }, [props.fileSubmissionData.drawings]);
 
   return (
     <div id="draw-step">
       <div id="draw" className="gameification-content">
         <div className="inner-container">
-          <img src={boyImg} alt="Boy Image" className="boy-img" />
+          <img src={boyImg} alt="A boy" className="boy-img" />
```
Comment (rhiannonstanford) [inline src/components/pages/Gameification/GameDrawStep.js:107]: Nice fix.  This is a better description.

File: src/components/pages/Gameification/GameDrawStep.js [hunk 2]
```diff
@@ -97,13 +97,13 @@ export default function GameDrawStep(props) {
     });
 
     setFileList(props.fileSubmissionData.drawings);
-  }, []);
+  }, [props.fileSubmissionData.drawings]);
```
Comment (rhiannonstanford) [inline src/components/pages/Gameification/GameDrawStep.js:100]: This resolves: **./src/components/pages/Gameification/GameDrawStep.js**
Line 100:6:   React Hook useEffect has a missing dependency: 'props.fileSubmissionData.drawings'

**Comment by ashtilawat** [comment]
No need to apologize! We don't want to delete the file from the repo completely. We just want to maintain the version that is on main already. See https://github.com/Lambda-School-Labs/scribble-stadium-fe/pull/186#pullrequestreview-775529017

File: src/components/common/UploadDocs.js [hunk 3]
```diff
@@ -114,11 +114,11 @@ export const UploadDocs = ({
     console.log({ fileList, filePreviews }, 'useEffect');
 
     // This is in case I have a filelist ready OUTSIDE this component
-    if (savedFileList != undefined) {
+    if (savedFileList !== undefined) {
       setFilePreviews(savedFileList);
       setFileList(savedFileList);
     }
-  });
+  }, [fileList, filePreviews, savedFileList]);
```
Comment (rhiannonstanford) [inline src/components/common/UploadDocs.js:121]: This line resolves: **./src/components/common/UploadDocs.js**
Line 113:3:   React Hook useEffect contains a call to 'setFilePreviews'

**Review by rhiannonstanford — DISMISSED** [review]
I was able to read through the code changes well and understand the errors you were resolving and how you resolved them through a very descriptive intro to this PR.  Nice work, just need to maintain the package-lock.json file from main.

File: src/components/pages/Gameification/GameDrawStep.js (hunk 1 above)
Comment (OniWarrior) [reply src/components/pages/Gameification/GameDrawStep.js:107]: Thank you.

**Comment by OniWarrior** [comment]
Ok I followed the instructions in the review and restored the package-lock.json file to what is in main.

**Review by ethankarp — CHANGES_REQUESTED** [review]
Nice job with these fixes! It looks like you'll need to rebase this in order to merge it. As a reminder, that's a "git pull" on the main branch, then a "git rebase main" on this branch.

**Comment by OniWarrior** [comment]
I didn't commit and push the file because I didn't want to make the same mistake twice.

**Comment by OniWarrior** [comment]
Ok I did a git pull on main and a git rebase main on the bug branch. Everything should be ready now.

**Comment by ethankarp** [comment]
Now if you commit and push, this will be merge-able

**Comment by OniWarrior** [comment]
The push wasn't successful.  The previous package-lock.json file on the remote branch was interfering with the restored main version. I've tried resolving the issue by following the instructions provided but it's not going through. I've created a brand new clean branch that I'm marking as redo at the end of the branch name. This is the only thing I can think of to resolve this issue.

**Pull request closed without merging**